File: app/src/main/java/com/example/vincentwang/rainbowtranslate/translate/TranslateContract.java
package com.example.vincentwang.rainbowtranslate.translate;

import com.example.vincentwang.rainbowtranslate.data.WordMain;
import com.example.vincentwang.rainbowtranslate.data.WordTotalInfo;
import com.example.vincentwang.rainbowtranslate.framework.BaseContract;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

import io.reactivex.Flowable;

/**
 * Created by vincentwang on 2017/8/9.
 */

public interface TranslateContract {
    interface View extends BaseContract.View{
        void showWordTranslateInfo(List<WordTotalInfo> wordTotalInfos);
    }
    interface Presneter{
        void loadWordAllInfo(String word);
    }
    interface Model{
        Flowable<ArrayList<WordTotalInfo>> getWordTranslateInfo(String word);
        Flowable<ArrayList<WordMain>> getWordMainPeriod(Calendar startDay, Calendar endDay);
        Flowable<ArrayList<WordMain>> getWordMainToday();
        Flowable<ArrayList<WordMain>> getWordMainAll();
    }
}

File: app/src/main/java/com/mvp/vincentwang/rainbowtranslate/translate/TranslateContract.kt
package com.mvp.vincentwang.rainbowtranslate.translate

import com.mvp.vincentwang.rainbowtranslate.framework.BaseContract
import com.mvp.vincentwang.rainbowtranslate.room.data.WordTotalInfo

/**
 * Created by vincentwang on 2017/8/9.
 */

interface TranslateContract {
    interface View : BaseContract.View {
        fun showWordTranslateInfo(word:String,wordTotalInfos: List<WordTotalInfo>)
    }

    interface Presneter:BaseContract.Presenter {
        fun loadWordAllInfo(word: String?)
        fun loadWordAllInfoByStore()
    }
}
